storage/src/main/java/fk/prof/storage/AsyncStorage.java

```diff
@@ -0,0 +1,27 @@
+package fk.prof.storage;
+
+import java.io.InputStream;
+import java.util.concurrent.Future;
+
+/**
+ * Interface for a simple async load/store storage.
+ * @author gaurav.ashok
+ */
+public interface AsyncStorage {
+
+    /**
+     * Store the content to the specified path. IOException while storing
+     * needs to be taken care of by the implementation.
+     * @param path
+     * @param content
+     */
+    void store(String path, InputStream content);
```

Comment:
Anyway, D42AsyncStorage is nothing but a simple wrapper over S3 client and executor service, both of which will be created once per app lifetime. So i dont see much issue in creating creating  D42AsyncStorage instance for different bucket (this instance may very well be created on the fly when needed).